client/sites-dashboard/components/sites-dashboard-v2/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { useI18n } from '@wordpress/react-i18n';
+import classNames from 'classnames';
+import { translate } from 'i18n-calypso';
+import { useCallback, useEffect, useState } from 'react';
+import {
+	DATAVIEWS_TABLE,
+	initialDataViewsState,
+} from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import { DataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces';
+import Layout from 'calypso/a8c-for-agencies/components/layout';
+import LayoutColumn from 'calypso/a8c-for-agencies/components/layout/column';
+import LayoutHeader, {
+	LayoutHeaderActions as Actions,
+	LayoutHeaderTitle as Title,
+} from 'calypso/a8c-for-agencies/components/layout/header';
+import LayoutTop from 'calypso/a8c-for-agencies/components/layout/top';
+import DocumentHead from 'calypso/components/data/document-head';
+import { useSiteExcerptsQuery } from 'calypso/data/sites/use-site-excerpts-query';
+import { SitesDashboardQueryParams } from 'calypso/sites-dashboard/components/sites-content-controls';
+import {
+	useShowSiteCreationNotice,
+	useShowSiteTransferredNotice,
+} from 'calypso/sites-dashboard/components/sites-dashboard';
+import { useDispatch } from 'calypso/state';
+import { setSelectedSiteId } from 'calypso/state/ui/actions';
+import DotcomPreviewPane from './site-preview-pane/dotcom-preview-pane';
+import SitesDashboardHeader from './sites-dashboard-header';
+import DotcomSitesDataViews from './sites-dataviews';
+
+// todo: we are using A4A styles until we extract them as common styles in the ItemsDashboard component
+import './style.scss';
+
+interface SitesDashboardProps {
+	queryParams: SitesDashboardQueryParams;
+}
+
+const SitesDashboardV2 = ( {
+	queryParams: { page = 1, perPage = 96, search, newSiteID },
+}: SitesDashboardProps ) => {
+	const { __ } = useI18n();
+	const dispatch = useDispatch();
+
+	const { data: liveSites = [], isLoading } = useSiteExcerptsQuery(
+		[],
+		( site ) => ! site.options?.is_domain_only
+	);
+
+	const { data: deletedSites = [] } = useSiteExcerptsQuery(
+		[],
+		( site ) => ! site.options?.is_domain_only,
+		'deleted'
+	);
+
+	const allSites = liveSites.concat( deletedSites );
+
+	useShowSiteCreationNotice( allSites, newSiteID );
+	useShowSiteTransferredNotice();
+
+	// Create the DataViews state based on initial values
+	initialDataViewsState.page = page;
+	initialDataViewsState.perPage = perPage;
```

Comment:
While the search param seems to work as expected on load, page seems to get reset in the dataview state. Just logging the props, initial state object, and the data views state object shows:

![Screenshot 2024-04-23 at 4 33 45 PM](https://github.com/Automattic/wp-calypso/assets/28742426/37aacd97-aeae-4aeb-b049-58c8215760d5)

Im not quite sure what is causing the rerender and reset yet and if its down at the gutenberg component level. There may be a similar issue with perPage as well, but there is a separate bug preventing that from being passed into the component from the queryParams correctly in the first place (always the default).

In general, I wonder how valuable having the search/filter/pagination states in the URL params actually is. It does make using the browser "back" button conveniently end you up back at the filtered list you were previously looking at (in the current dotcom sites dashboard), but I wonder if there really is much value in this. cc @davemart-in if you have any thoughts or insight on this.  This question is probably less meaningful for this PR and more for subsequent PRs to enable the search/filter/pagination functionality.

Reply:
I guess the only other helpful side effect is that you can share urls with others that would take them directly to the view that you're referring to.

At the very least, I suspect this is not a blocker and something we could add a separate issue for and circle back to after the i2 launch.